### DISCUSSION TITLE
Utm stuck on startup.nsh

### DISCUSSION THREAD

**Discussion by SalishKumar** [utmapp/UTM#3470] [UTM Help (macOS)] [answered]
I want to install parrot os, i have configured the machine but when I starts it doesn't start installation, it shows startup.nsh

**Configuration**
* 3.0.2: 
* monterey: 
* m1 
<img width="842" alt="Screen Shot 2022-01-12 at 11 20 17 AM" src="https://user-images.githubusercontent.com/54025715/149074587-35fc48a5-64f2-4676-99fc-b77a3fb6eb35.png">

**Comment by conath** [comment]
Hi. I moved this question to the discussion section since it’s about a guest OS and not about the UTM app itself.

Please download my “Ubuntu Desktop 20.04 (x86_64)” template from [my UTM website](https://chrisp.cafe/UTM/) and try installing Parrot OS in that VM. You can rename it, of course.

**Reply by SalishKumar** [reply]
<img width="810" alt="Screen Shot 2022-01-12 at 10 25 47 PM" src="https://user-images.githubusercontent.com/54025715/149190679-b917191f-9e72-4dcd-87ae-a82100441ccb.png">

@conath STILL SAME output

**Reply by conath** [reply]
Please add a screen recording of how you are trying to install Parrot OS. Start the recording at the Desktop before you open UTM, please. Here’s [how to make a screen recording on Mac](https://support.apple.com/en-us/HT208721).

**Reply by SalishKumar** [reply]
@conath  please have a look "https://drive.google.com/file/d/1NU7ZNYNVLM4_M3_55sJkvPOUcZkypbiw/view?usp=sharing"

**Reply by conath** [reply]
Ok. You don’t need the Ubuntu ARM VM. Did you choose the installer iso into the CD drive of the VM?

**Comment by osy** [comment]
Common mistake is using the wrong architecture. Check that it isn’t the case here.

**Reply by SalishKumar** [reply]
<img width="664" alt="Screen Shot 2022-01-12 at 10 22 17 PM" src="https://user-images.githubusercontent.com/54025715/149190149-cbe37248-7b64-45e4-8761-e7d0bc8571d9.png">
@osy I think architecture is correct

**Reply by osy** [reply]
It clearly isn't. Look at your screenshot. Your ISO is **amd64** while the architecture is **arm64**

**Reply by conath** [reply]
It’s incorrect, for an installer with “amd64” choose x84_64 in UTM please.

**Reply by thecodemonster1** [reply]
> It’s incorrect, for an installer with “amd64” choose x84_64 in UTM please.

Thank You So much brother, This is helped me also😊

**Comment by SalishKumar** [comment, marked as answer]
@conath @osy  thanks for your help, i selected wrong architecture and iso file. 
following this link solved my issue "https://mac.getutm.app/gallery/ubuntu-20-04"

**Comment by SalishKumar** [comment]
<img width="541" alt="Screen Shot 2022-01-13 at 12 45 40 AM" src="https://user-images.githubusercontent.com/54025715/149210788-c59d782c-176f-4a38-b1a6-971476c08ffa.png">

@conath ubuntu stuck on boot after installation

**Reply by conath** [reply]
Sounds like #3344. Please close the VM window and remove the CD image, then start the VM again.

**Comment by pipiche38** [comment]
I have the similar issue, after imported a VM which was created under VirtualBox.
I did an export on VirtualBox and Import on UTM

No issue to do a life recording, but there is not much to see.
<img width="803" alt="Screenshot 2022-02-01 at 11 09 01" src="https://user-images.githubusercontent.com/8291674/151949612-67cd75bb-2d1f-4b6f-8093-956b6fd9163b.png">

**Reply by conath** [reply]
Please try disabling UEFI boot by editing the VM, the option is in System => Show Advanced Settings.

**Reply by pipiche38** [reply]
It was enabled on VirtualBox, and if I disable on UTM side, the boot fails with No bootable device.

**Reply by pipiche38** [reply]
The VirtualBox config was
PIIX3
SATA AHCI for Storage

**Reply by conath** [reply]
Try changing the system to “Standard PC (q35…) (pc-q35-6.2)”. If you choose “q35” system (which is the default) and “IDE” drive interface, it will be connected as SATA AHCI to the guest.

**Reply by pipiche38** [reply]
Unfortunately not better.
On the VirtualBox the Storage Controller Type is: AHCI and the disk is defined as Solid-state Drive on SATA Port 0

**Reply by conath** [reply]
Please commend with a screenshot of your imported drive selected in the VM config window, and also add a debug log from when you start the VM. [How to get Debug Log](https://github.com/utmapp/UTM/wiki/Troubleshooting-a-VM#how-to-get-debug-log)

**Reply by pipiche38** [reply]
<img width="459" alt="Screenshot 2022-02-01 at 11 46 32" src="https://user-images.githubusercontent.com/8291674/151955054-c28ae066-0123-400b-8866-220f40c48f93.png">


[debug.log](https://github.com/utmapp/UTM/files/7977747/debug.log)

**Reply by conath** [reply]
I see, you imported the OVA file. Unfortunately UTM doesn’t know to extract this file for you, you need to do this manually. Here is how to import the hard drive file from the OVA: https://github.com/utmapp/UTM/issues/3540#issuecomment-1023190945

**Reply by fortran01** [reply]
Not sure what's happening. I switched to the version mentioned above “Standard PC (q35…) (pc-q35-6.2)” and switched back to the default `(Standard PC (Q35 + ICH9, 2009) (alias of pc-q35-10.0) (q35)`. It worked.

**Comment by ststst4** [comment]
I have a similar issue when I tried to boot a windows vm can you help me out?

**Reply by conath** [reply]
Need more info. How did you create the VM, where did you get the installer/image, which exact version of Windows is it (10,11? ARM or X64?) 
It would be best if you posted your question and the info on the UTM Discord, since the community support is more active there: https://discord.gg/UV2RUgD

**Reply by ststst4** [reply]
I realized that i didn't download an iso. so  i did that, now what do i do next.

**Reply by ststst4** [reply]
*?

**Reply by ststst4** [reply]
i was wrong its still not working

**Reply by conath** [reply]
Next, please answer the questions:

> Need more info. How did you create the VM, where did you get the installer/image, which exact version of Windows is it (10,11? ARM or X64?) 
> 
> It would be best if you posted your question and the info on the UTM Discord, since the community support is more active there: https://discord.gg/UV2RUgD

**Reply by ststst4** [reply]
I made the vm on the arm64 architecture and with the qemu 7.0 arm vm

**Reply by ststst4** [reply]
i got the windows 10 iso from microsofts website.

**Reply by conath** [reply]
Alright so you've got an X64 ISO there and made an "ARM64" VM. This is not compatible, you need to make a "x86_64" emulated VM, or you need to get a Windows 11 ARM file from Microsoft (provided in VHDX format, which UTM can convert).

**Reply by ststst4** [reply]
I did that and it still didn't work

**Reply by conath** [reply]
If you want to make a fast Windows VM on your Mac, follow [this guide](https://mac.getutm.app/gallery/windows-10). If you want a slow (because emulating on M1), "classic" x86_64 Windows VM on your Mac, follow this [video guide](https://www.youtube.com/watch?v=KsNoGPczavw).

**Comment by egladman** [comment]
For what it's worth....I got a similar error on startup when loading a Fedora variant. I had to disable uefi boot.

**Comment by vysecurity** [comment]
I'm trying to make it emulate x64 Ubuntu, but it keeps saying startup.nsh or if I disable UEFI boot it gives me no bootable device. What do I do?

**Reply by conath** [reply]
Please check the guide: https://mac.getutm.app/gallery/ubuntu-20-04

For X64 Ubuntu you need to choose "Emulate" instead of "virtualize".

**Comment by vysecurity** [comment]
That’s ARM though. I got AMD working now but yah it’s mega slow.

**Comment by ny8wolf** [comment]
Same issue on M1 - ARM64 Win11 ISO from the link on UTM guide. 
![image](https://user-images.githubusercontent.com/123138586/213610643-a692426c-2277-4dd6-8fc4-930149530c4b.png)

**Reply by YassineIdr02** [reply]
same problem... did u find a solution??

**Reply by Victree-Mthombeni** [reply]
I have managed to solve this issue with windows 10

I followed the following video: https://www.youtube.com/watch?v=8gIa8guWV9o

However, i had to repeatedly enter the second command multiple times.

**Reply by EthanDM** [reply]
Thanks @Victree-Mthombeni, that worked for me with Windows 11 on M1. I seemed to have a little issue too so I ran the following sequence of commands:

`cd EFI`
`cd BOOT`
`BOOTAA64.EFI`

and I had to run it twice to get it to work.

**Reply by invokermoon** [reply]
same issue

**Reply by Qhilm** [reply]
the following command is not working for me: `FS0:\EFI\BOOT\BOOTAA64.EFI`

It does reboot, but it goes back to the `Shell>` prompt... (I did try multiple times).

* UTM 4.3.5 (87)
* macOS 13.5
* ISO grabed using CrystalFetch: 22621.1_MULTI_ARM64_EN-US.ISO

**Reply by Qhilm** [reply]
Ignore, I had to press a key when it said "press any key to boot from CD or DVD" (duh...). It worked! Thanks.

Command can be simplified in a one liner: `FS0:\EFI\BOOT\BOOTAA64.EFI`

**Reply by hontz1** [reply]
i had downloaded 2 iso of win 11 arm. tiny is working without problems. the other ran also into this error, bootaa64.efi also not working. I should try the official.

**Reply by rogeliogomezusb** [reply]
That u, that command really work for me. I had the same issue @Qhilm

**Comment by 0xvoila** [comment]
Thank you, it worked for me as well.

**Comment by DJAKLSJDLKA** [comment]
Who can help me with this problem, or is it unsolvable?
[
![Uploading Screenshot 2023-11-03 at 10.03.51 AM.png…]()
](url)

**Comment by emrekocan** [comment]
I get this warning when I try to install utm

Macbook m1

**Reply by emrekocan** [reply]
![Uploading IMG_2649.png…]()

**Comment by mijorus** [comment]
Just sharing how I fixed it:

1. Go into the VM settings > QEMU > Reset EFI Variables
2. Boot the machine
3. Press Enter repeatedly until you see the white spinner under the UTM logo

**Reply by motorzen** [reply]
This was the fix for me - I didn't actually change/reset any QEMU settings, but the clue for me was to simply hit a key when "press any key to boot from CD or DVD" displays in the process the first time. 

It goes against what I expected, which is - I am installing from an image on a hard drive, rather than a CD/DVD.

**Reply by aysilsimgekaracan** [reply]
Thank you, this worked for me too

**Comment by kokoskiwi** [comment]
hi how to solve this? i downloaded Tails .iso and am stuck in this terminal too....

**Comment by sos5** [comment]
help SOS (MAC)
I want to install  ubuntu by utm , i have configured the machine but when I starts it doesn't start installation, it shows startup.nsh
![Annotation 2024-02-29 090356](https://github.com/utmapp/UTM/assets/161668241/1dafa030-c564-46e4-9571-da71008b6a1a)

![Annotation 2024-02-29 090356](https://github.com/utmapp/UTM/assets/161668241/1dafa030-c564-46e4-9571-da71008b6a1a)

**Comment by hontz1** [comment]
for win 11 UTM i used some bad iso, found on web (archive). then I got these problems, too. because crystal fetch aborted. downloading the official iso might solve your problems, too. I used https://github.com/mattieb/download-windows-esd
with this tool, you can resume download, if your internet-connection times out.

**Comment by silverlining2242** [comment]
type "exit" and then choose continue. it will start to install windows in GUI

**Comment by reiz166u90** [comment]
https://github.com/utmapp/UTM/assets/99326853/a71f9d1b-f2f4-4a2c-942f-cac92b7ebe40

help me 
macos version sonoma 14.3.1
chip Apple m2

**Reply by silverlining2242** [reply]
I type "exit" and press enter to escape the issue of stuck in the UEFI shell while booting win11 installation.
btw, when set up the virtual machine disk for win11 make sure assign enough space, otherwise it will create another pain to expand the disk size...

**Reply by MoThor25** [reply]
THANK YOU GODDDD

**Comment by scott-joe** [comment]
Sorry if I'm beating a dead horse here, but I've downloaded Fedora 40 ARM64, which comes as a `Fedora-Workstation-40-1.14.aarch64.raw.xz`. For whatever reason, it's not available as an ISO and the various tools I've attempted to use to convert the `raw.xz` file into an `.iso` were unsuccessful. When loading it into UTM under ARM64 Linux, it gives me that UEFI Interactive Shell people are showing...

Thanks in advance.

**Comment by rv2215** [comment]
Trying to Run Windows 11 and it is stuck in SHELL mode.

<img width="901" alt="Screenshot 2024-05-15 at 4 41 22 PM" src="https://github.com/utmapp/UTM/assets/169946580/802fbc90-6917-4f15-9e0a-245150ab5a62">

**Comment by Behinder** [comment]
I tried to boot SteamDeckOS on emulating x86 architecture and it's the same. Tested on machine 7.2,7.1,7.0,6.2
![Uploading Zrzut ekranu 2024-06-25 o 10.45.59.png…]()

**Comment by Gnomecromancer** [comment]
Same issue but I’m 90% sure my architecure is correct
<img width="1343" alt="Screen Shot 2024-10-08 at 3 35 57 PM" src="https://github.com/user-attachments/assets/405103ef-168f-4204-9711-97614b132ad5">

**Comment by rawrbw** [comment]
For anyone like me that didnt press a key at this screen, thats your solution... ;)
![press-any-key-to-boot-from-cd-dvd_thumb-221612888](https://github.com/user-attachments/assets/7b721e18-4ae0-48fd-b5d8-523c96a50453)

**Reply by triumph1-ably** [reply]
so it was it.. I just had to press enter (or any key)!!! thanks!

**Comment by leo00-opes** [comment]
Me too windows 11 vm 
Macbook Air M1,2020 , macos sequoia 15.4

**Reply by thecodemonster1** [reply]
Make sure you’re using the ARM64 version of Windows—Apple Silicon doesn’t support x86/x64.

It works, but the MacBook Air can get quite hot (no cooling fan), so it’s okay for light, occasional use.

**Comment by pavlotsyhanok** [comment]
If your Windows VM on UTM isn’t booting and just shows a blank screen or doesn’t seem to load the installer, here’s the fix:

As soon as the VM window opens, immediately press any key (e.g., Enter).

The Windows installation ISO only gives a very short window to detect a key press (“Press any key to boot from CD or DVD…”). On Apple Silicon Macs, the display sometimes lags slightly, so by the time you see that message, it might already be too late.

Once I pressed Enter right after the VM window appeared, the installer booted normally.
No other settings needed to be changed, I kept all defaults.

**Comment by dhlavaty** [comment]
Any tips how to run Raspberry Pi OS ( it is ARM64 downloaded from `https://downloads.raspberrypi.com/raspios_arm64/images/raspios_arm64-2025-11-24/2025-11-24-raspios-trixie-arm64.img.xz` ) on Macbook M4 Apple Silicon ? I've tried .img.xz and also uncompressed .img file. Tried all tips found here and other discussions in this github repo. I always end up in UEFI Shell.

**Comment by blp92** [comment]
I'm having the same issue of being perpetually stuck in the UEFI shell but nothing I've tried has fixed it. I'm trying to install a Windows 10 VM. I'm fairly confident I have the right settings. I've tried to exit and continue, reset the UEFI variables and reload, press enter immediately upon starting the VM and at every single other possible interval, but it never ends anywhere but the UEFI shell
<img width="641" height="392" alt="Screenshot 2026-02-27 at 14 26 59" src="https://github.com/user-attachments/assets/0759ff95-44a6-4f2f-b375-0b7662912782" />

**Reply by blp92** [reply]
I've tried with multiple ISOs (all arm64) and it still just lands in UEFI shell

**Comment by hulkthesmart** [comment]
<img width="980" height="784" alt="48EE31F9-E474-477E-9AB0-B2CE74A4AE9A" src="https://github.com/user-attachments/assets/b5513cc8-a93a-4d15-b535-00ff57d97560" />
i also faced the same issue.. Just click "Esc" immediately when the attached screenshot screen appears, then the windows will start installation

**Comment by yking112232-tech** [comment]
yea hi have same issue but on neo
<img width="801" height="598" alt="image" src="https://github.com/user-attachments/assets/3379f2b8-8f5c-48d3-a3d5-1eeb8926c3d5" />
